catalog/core/catalog-core-backupplugin/src/test/java/ddf/catalog/backup/CatalogBackupPluginTest.java

```diff
@@ -470,14 +336,65 @@ private String getMetacardPath(String metacardId, int subDirLevels) {
         return builder.toString();
     }
 
-    private Metacard readMetacard(File file) throws ClassNotFoundException, IOException {
-        Metacard metacard = null;
+    private Metacard readMetacard(File file) {
 
+        Metacard metacard = null;
         try (FileInputStream fis = new FileInputStream(file);
                 ObjectInputStream ois = new ObjectInputStream(fis)) {
             metacard = (Metacard) ois.readObject();
+        } catch (ClassNotFoundException | IOException e) {
+            fail();
         }
 
         return metacard;
     }
+
+    private File getFileFor(String metacardId, int subDirLevels) {
+        return new File(rootBackupDir.getRoot()
+                .getAbsolutePath() + getMetacardPath(metacardId, subDirLevels) + metacardId);
+    }
+
+    private void assertFilesDoNotExist(String[] metacardIds, int subDirLevels) {
+        assertFiles(Boolean.FALSE, metacardIds, subDirLevels);
+
+    }
+
+    private void assertFilesExist(String[] metacardIds, int subDirLevels) {
+        assertFiles(Boolean.TRUE, metacardIds, subDirLevels);
+    }
+
+    private void assertFiles(Boolean test, String[] metacardIds, int subDirLevels) {
+        for (String id : metacardIds) {
+            assertThat(getFileFor(id, subDirLevels).exists(), is(test));
+        }
+    }
+
+    private void createBackupFilesFor(String[] cardIds, int level) {
+
+        getPlugin(level).backup(getMetacards(cardIds, BASE_OLD_TITLE));
+        assertFilesExist(cardIds, level);
+
+    }
+
+    private void createBackupFilesFor(String[] cardIds) {
+        int level = 3;
+        getPlugin(level).backup(getMetacards(cardIds, BASE_OLD_TITLE));
+        assertFilesExist(cardIds, level);
+
+    }
+
+    private CatalogBackupPlugin getPlugin() {
+
+        return getPlugin(3);
+    }
+
+    private CatalogBackupPlugin getPlugin(int level) {
+        CatalogBackupPlugin plugin = new CatalogBackupPlugin();
```

Comment:
Would it be better in inject the PluginBatchExecutor so you can mock it out in the unit tests?